### PR TITLE
Add doc to describe internal metadata (_properties); Add advanced query examples

### PR DESCRIPTION
**Pull request by erkangz**

### CODE REVIEW

File: docs/metadata.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+> - `deletedOn`
+> - `startedOn`
+> - `stoppedOn`
+
+## State and source related metadata
+
+| Property     | Type      | Description                                        |
+| ------------ | --------- | -------------------------------------------------- |
+| `_deleted`   | `boolean` | Indicates whether a resource was deleted from JupiterOne graph/CMDB. This typically means the resource was recently deleted from the provider source environment.
+| `_latest`    | `boolean` | Indicates if an entity/relationship is the latest version.
```
Comment (austin-rausch) [inline docs/metadata.md:39]: I think this property has been deprecated and I'm surprised to see it still on some entities.

File: docs/metadata.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+## Integration specific metadata
+
+_The following metadata only exists on resources created via an integration._
+
+| Property            | Type                | Description                                        |
+| ------------------- | ------------------- | -------------------------------------------------- |
+| `_integrationClass` | `string | string[]` | Class of the integration that represents the category or domain of the service provider.  For example: `CSP`, `IdP`, `EDR`, `MDM`, etc.
+| `_integrationType`  | `string`            | Type of the integration. Typically the service provider name. For example: `aws`, `google`, `azure`, `okta`, `knowbe4`, `vmware`, etc.
+| `_integrationName`  | `string`            | User-provided friendly name of the integration instance.
+| `_integrationDefinitionId` | `string`     | Internal UUID that identifies the integration.
```
Comment (austin-rausch) [inline docs/metadata.md:52]: Clarification for this, the current wording makes it hard to distinguish from `_integrationInstanceId`:
`Internal UUID that identifies the definition for this integration, E.G. AWS, Azure, etc.`

File: queries/adv-example-resources-not-tagged.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
+# Query resources that do _not_ have a certain tag
+
+Tagging is highly recommended for resources in cloud environments such as AWS,
+Azure, and GCP. For example, it is common to use tags to track `cost-center` or
```
Comment (austin-rausch) [inline queries/adv-example-resources-not-tagged.md:4]: So this tag came directly from a question in the community slack, do we want to use that concrete example or use something more general? I'll leave this to you, but `cost-center` doesn't seem like a super universal tag to me.

Comment (erkangz) [reply queries/adv-example-resources-not-tagged.md:4]: it's common for large orgs/enterprises

Comment (henrymgarrett) [reply queries/adv-example-resources-not-tagged.md:4]: FWIW we used cost-center tagging at my previous company

**Pull request merged**